strawberryfields/io/xir_io.py

```diff
@@ -0,0 +1,294 @@
+
+            params = {}
+            if cmd.op.p:
+                # argument is quadrature phase
+                a = cmd.op.p[0]
+                if a in getattr(prog, "loop_vars", ()):
+                    params["phi"] = a.name
+                else:
+                    params["phi"] = a
+
+            # special case to take into account 'select' keyword argument
+            if cmd.op.select is not None:
+                params["select"] = cmd.op.select
+
+            if name == "MeasureFock":
+                # special case to take into account 'dark_counts' keyword argument
+                if cmd.op.dark_counts is not None:
+                    params["dark_counts"] = cmd.op.dark_counts
+        else:
+            if kwargs.get("add_decl", False):
+                if name not in [gdecl.name for gdecl in xir_prog.declarations["gate"]]:
+                    params = [f"p{i}" for i, _ in enumerate(cmd.op.p)]
+                    gate_decl = xir.Declaration(
+                        name, type_="gate", params=params, wires=tuple(range(len(wires)))
+                    )
+                    xir_prog.add_declaration(gate_decl)
+
+            params = []
+            for i, a in enumerate(cmd.op.p):
+                if sfpar.par_is_symbolic(a):
+                    # try to evaluate symbolic parameter
+                    try:
+                        a = sfpar.par_evaluate(a)
+                    except sfpar.ParameterError:
+                        # if a tdm param
+                        if a in getattr(prog, "loop_vars", ()):
+                            a = a.name
+                        # if a pure symbol (free parameter), convert to string
+                        elif a.is_symbol:
+                            a = a.name
+                        # else, assume it's a symbolic function and replace all free parameters
+                        # with string representations
+                        else:
+                            symbolic_func = a.copy()
+                            for s in symbolic_func.free_symbols:
+                                symbolic_func = symbolic_func.subs(s, s.name)
+                            a = str(symbolic_func)
+                elif isinstance(a, Iterable):
+                    # if an iterable, make sure it only consists of lists and Python types
+                    a = _listr(a)
+                params.append(a)
+
+        op = xir.Statement(name, params, wires)
+        xir_prog.add_statement(op)
+
+    return xir_prog
+
+
+def _listr(mixed_list):
+    """Casts a nested iterable to a list recursively"""
+    list_ = []
+
+    for l in mixed_list:
+        if isinstance(l, Iterable):
+            list_.append(_listr(l))
+        else:
+            if isinstance(l, Decimal):
+                list_.append(float(l))
+            elif isinstance(l, xir.DecimalComplex):
+                list_.append(complex(l))
+            else:
+                try:
+                    list_.append(l.item())
```

Comment:
What type is this for? Should there be an accompanying comment?

Reply:
If `l` is a NumPy type (e.g., `np.int64`) `l.item()` will extract the internal Python type object (e.g., `int`).

Added a comment!